### ISSUE TITLE
Proposal: Merge Skeleton::setPositions() and Skeleton::setPositionSegment()

### ISSUE THREAD

**Issue by mxgrey** [dartsim/dart#383]
Rather than having two differently named functions for essentially the same purpose, I think it would make sense to just have one overloaded function:

```
void Skeleton::setPositions(const Eigen::VectorXd& _positions);
void Skeleton::setPositions(const Eigen::VectorXd& _positions, const std::vector<size_t>& _indices)
```

This proposal also applies to all the other functions that have a "Segment" version, such as ``Skeleton::setVelocitySegment()`` and ``Skeleton::setAccelerationSegment()``.

**Comment by jslee02** [comment]
Sounds good to me.

**Comment by mxgrey** [comment]
Actually, to match the existing pattern of ``setPosition(size_t, double)``, I'm thinking we should make it ``setPositions(const std::vector<size_t>&, const Eigen::VectorXd&)``.

**Comment by jslee02** [comment]
Matching pattern looks good as well. How about applying this change to `getPositionSegment()`?

**Comment by mxgrey** [comment]
Yep, I intend to do the same with every ``get[X]Segment`` function as well.

**Comment by jslee02** [comment]
Great!

**Comment by mxgrey** [comment]
This issue has been resolved in the memory_management branch #369
